src/keeper/contracts/ContractBase.ts

```diff
@@ -1,24 +1,56 @@
+import { RelayProvider } from '@opengsn/provider'
+import { TransactionResponse } from '@ethersproject/abstract-provider'
 import ContractHandler from '../ContractHandler'
 
 import Account from '../../nevermined/Account'
 import { ContractEvent, EventHandler, SubgraphEvent } from '../../events'
 import { Instantiable, InstantiableConfig } from '../../Instantiable.abstract'
 import { KeeperError } from '../../errors'
 import { ContractReceipt, ethers } from 'ethers'
-import { TransactionResponse } from '@ethersproject/abstract-provider'
 import BigNumber from '../../utils/BigNumber'
 
+export interface MetaTxParameters {
+    paymasterAddress: string
+    wallet: ethers.Wallet
+}
+
 export interface TxParameters {
     value?: string
     gasLimit?: BigNumber
     gasMultiplier?: number
     gasPrice?: string
     maxPriorityFeePerGas?: string
     maxFeePerGas?: string
+    meta?: MetaTxParameters
     nonce?: number
     progress?: (data: any) => void
 }
 
+export interface JsonRpcPayload {
+    jsonrpc: string;
+    method: string;
+    params: any[];
+    id?: string | number;
+}
+
+export interface JsonRpcResponse {
+    jsonrpc: string;
+    id: number;
+    result?: any;
+    error?: string;
+}
+
+class Web3ProviderWrapper {
+    provider: ethers.providers.JsonRpcProvider
+    constructor(provider: ethers.providers.JsonRpcProvider) {
+        this.provider = provider
+    }
+    send(payload: JsonRpcPayload, callback: (error: Error | null, result?: JsonRpcResponse) => void): void {
+        const id = typeof payload.id === 'string' ? parseInt(payload.id) : payload.id
+        this.provider.send(payload.method, payload.params).then(result => callback(null, {jsonrpc: payload.jsonrpc, id, result}))
+    }
+}
+
```

Comment:
I'm assuming that all this relates to opengsn. If so this should have better names and maybe be on a separate source file